### ISSUE TITLE
PR description: suggestions

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode-pull-request-github#2401]
Refs: https://github.com/microsoft/vscode/issues/114967

Would be cool if in the PR create flow in the Descripiton I could use suggestions to get offered issue numbers, user names etc.. Like we have in other parts of the GH UI in VS Code.
My use case is that in the PR descripiton I would usually say "fixes #ISSUE_NUMBER"

**Comment by tanhakabir** [comment]
+1 for auto-complete for issue #'s and @'s

**Comment by bamurtaugh** [comment]
cc @alexr00 wanted to raise awareness that we received github.dev feedback that commenting could be more powerful in GHPRI, including suggestions/auto-converting/hints for issues and usernames, emojis.

**Comment by alexr00** [comment]
@bamurtaugh commenting in the comments widget has suggestions already. Commenting in the PR overview editor is done in a webview, so we wont ever show suggestions.

**Comment by bamurtaugh** [comment]
Thanks for the context @alexr00! I see where `@` works to suggest a username in the comment widget. Should `#` suggest an issue or PR number, or `:` work to suggest an emoji?

Suggestion experience I see on gh.com:

![image](https://user-images.githubusercontent.com/25310137/141820829-0c9a74a3-cd1c-4f51-af4f-d56ae0f8b8bf.png)

![image](https://user-images.githubusercontent.com/25310137/141821189-52455981-cf7b-49f5-90c4-643713f81b34.png)

What I see in GHPRI comment widget:

![image](https://user-images.githubusercontent.com/25310137/141820960-6366d16e-9059-4c6c-b7a1-9551c124ee04.png)

![image](https://user-images.githubusercontent.com/25310137/141821231-824aa993-fba9-4d38-8af1-9ae3ee9cd60a.png)

> Commenting in the PR overview editor is done in a webview, so we wont ever show suggestions.

Or perhaps if we won't ever show suggestions in the webview, enhancing certain aspects of the comment widget is the outstanding aspect of this feature request, and my comment above is stating the already-known request? Although it sounds like this issue is more oriented around the description.

**Comment by alexr00** [comment]
I think most operating systems have their own emoji shortcut, so having a VS code specific one feels like overkill/duplication. What do you think?

`#` to trigger an issue suggestion does work, just not on the first character of the line in markdown. I have pushed a tweak to this to treat comments differently (and to show the completions when the suggest is invoked i.e., not trigger character). You should see this improved in the next nightly build!

**Comment by bamurtaugh** [comment]
> I think most operating systems have their own emoji shortcut, so having a VS code specific one feels like overkill/duplication. What do you think?

That's a good point - issue suggestion/completion is likely the more important component here.

***

I'm trying # in various comment combinations (comments on a specific file, PR overview page, first line in a comment, later lines in a comment), and I don't seem to be getting suggestions in github.dev with GHPRI nightly. Should I be seeing it now?

Example on https://github.dev/bamurtaugh/ExpressApp/pull/5:

![image](https://user-images.githubusercontent.com/25310137/143921967-e99a462c-9295-4113-92f3-86353a46fd6e.png)

On https://github.com/bamurtaugh/ExpressApp/pull/5/files: 

![image](https://user-images.githubusercontent.com/25310137/143922052-1721235f-b3b6-40ac-8992-23bcc5cc0500.png)

Please let me know if I'm missing something!

**Comment by alexr00** [comment]
@bamurtaugh looks like there were a few more issues that could happen. I've pushed a fix and queued a new pre-release build. Note that GHPRI now uses VS Code's built in pre-release support, and the old Nightly build has been deprecated (as of yesterday).

**Comment by bamurtaugh** [comment]
@alexr00 I now have only GHPRI stable enabled, `v0.32.0`. Should I see this change live when I go to https://github.dev/bamurtaugh/ExpressApp/pull/5 and type # in any comment area?

**Comment by alexr00** [comment]
@bamurtaugh you need to install the pre-release version, which can be done from the extensions view. GHPRI has a "switch to pre-release" (or something similar) button.

**Comment by bamurtaugh** [comment]
Thank you! I'm now on `v0.33.2021120709` and see "Pre-Release" next to the version number.

![image](https://user-images.githubusercontent.com/25310137/145099601-7efd6041-6a71-49a8-a0df-a795c5444b4f.png)

I apologize if I'm still doing something wrong, but when I type # in a comment (i.e. at https://github.dev/ataridotcom/gh-static-e2e/tree/test126), I don't see completions.

![image](https://user-images.githubusercontent.com/25310137/145099543-10794698-ed62-4439-9bcd-65892656b7d9.png)

**Comment by alexr00** [comment]
@bamurtaugh looks like that was due to the issues list not being populated. I've pushed a fix.

**Comment by bamurtaugh** [comment]
Thanks @alexr00! Should I see the change live in Pre-Release GHPRI `v0.33.2021120809` when I type a comment like the above?

**Comment by alexr00** [comment]
No, that one is from before the change. The new stable release, 0.34.0, has the fix, as do the 0.35.* pre-release versions.

**Comment by bamurtaugh** [comment]
Thanks for clarifying - I tried `v0.35.2021120909` pre-release and `v0.34.0` Stable, and I don't seem to have any different behavior when I type `#` in a comment. Any suggestions or things I should do differently?

**Comment by alexr00** [comment]
So this I cannot repro. Are issues showing in the issues view for you?

**Comment by bamurtaugh** [comment]
I do see issues and PRs in the GHPRI view on the left, but typing `#` in a comment does not suggest them (even if I start typing numbers after the `#` too):

![image](https://user-images.githubusercontent.com/25310137/145610606-16dfee50-12c0-432c-90d6-816282dbcc42.png)

I can successfully hover over the issues to get more info:

![image](https://user-images.githubusercontent.com/25310137/145610777-8d114776-28d8-4965-b03a-a2198d1e5364.png)

I tested with https://github.dev/bamurtaugh/ExpressApp/pull/5.

**Comment by alexr00** [comment]
Let's move over to https://github.com/microsoft/vscode-pull-request-github/issues/3182

**Comment by bamurtaugh** [comment]
Tested after seeing #3182, and looks like it's working great for me now!

![image](https://user-images.githubusercontent.com/25310137/145847747-a929e83e-e37e-4ec0-b3c0-f888151efd72.png)

Thanks so much for your continued help here, @alexr00!

**Comment by isidorn** [comment]
@alexr00 this is my biggest pain point when creating a PR from the GitHub PR extension. I need suggestions in order to ping the right people (I do not know the exact GitHub ids). 

> Commenting in the PR overview editor is done in a webview, so we wont ever show suggestions.

Can we embed a monaco editor inside the webview? Would that allow us to show suggestions? Would it bring other benefits?

**Comment by alexr00** [comment]
@isidorn this would be a huge undertaking and when I have brought it up it was not recommended. I can investigate further, but what if we had a simple "pop into editor" button? Then you can edit your PR description in a VS Code editor. This would be _much_ better than any embedded monaco editor experience we could offer.

**Comment by isidorn** [comment]
@alexr00 that might work. I agree that we should investigate that first since it is low-cost.

fyi @hbons

**Comment by hbons** [comment]
@isidorn me and @alexr00 brainstormed on this a little bit. We want to try this: show a hint after you type @ or # that will then open the QuickPick and you can then insert users and issue numbers from there.

**Comment by isidorn** [comment]
I would love to try out that flow :)

**Comment by alexr00** [comment]
> We want to try this: show a hint after you type @ or # that will then open the QuickPick and you can then insert users and issue numbers from there.

I talked about this at stand up and it was not popular. 

There was also very little enthusiasm for the headache that is embedding a monaco editor here.

**Comment by wesleyboar** [comment]
Maybe it would be easier to implement if the PR description was editable via a file edit pane, instead of within a limited editor.

- i.e. #6606

**Comment by alexr00** [comment]
No matter where the webview is, we have the same problem.

**Comment by ArtRodriguez** [comment]
I'm also affected by this in the GitHub Pull Requests extension for VS Code. When leaving a general comment on an existing PR, the Leave a comment text area in the PR overview doesn't offer user suggestions when I type @, so I can't search for and select a teammate to mention.
I'd expect typing @ (followed by part of a name or username) to show matching users and let me insert the selected @username. Please include the PR overview's comment box in the scope of this feature.
